src/solc_0.8.13/raffle/GenericRaffle.sol

```diff
@@ -0,0 +1,407 @@
+        require(_sandOwner != address(0x0), "Sand owner is zero address");
+        require(_maxSupply > 0, "Max supply should be more than 0");
+        sandOwner = _sandOwner;
+        signAddress = _signAddress;
+        maxSupply = _maxSupply;
+
+        emit ContractInitialized(baseURI, _name, _symbol, _sandOwner, _signAddress, _maxSupply);
+    }
+
+    function setupWave(
+        uint256 _waveType,
+        uint256 _waveMaxTokens,
+        uint256 _waveMaxTokensToBuy,
+        uint256 _waveSingleTokenPrice,
+        address _contractAddress,
+        uint256 _erc1155Id
+    ) external onlyOwner {
+        require(_waveMaxTokens <= maxSupply, "_waveMaxTokens should not exceed maxSupply");
+        require(_waveType < 3 && _waveMaxTokens > 0 && _waveMaxTokensToBuy > 0, "Invalid configuration");
+        if (_waveType != 0) {
+            require(_contractAddress != address(0x0), "Invalid contract address");
+            require(_contractAddress.isContract(), "Contract address must be that of a contract");
+        }
+        require(_waveMaxTokensToBuy <= _waveMaxTokens, "Invalid supply configuration");
+
+        waveType = _waveType;
+        waveMaxTokens = _waveMaxTokens;
+        waveMaxTokensToBuy = _waveMaxTokensToBuy;
+        waveSingleTokenPrice = _waveSingleTokenPrice;
+        waveTotalMinted = 0;
+        contractAddress = _waveType == 0 ? address(0x0) : _contractAddress;
+        erc1155Id = _waveType == 2 ? _erc1155Id : 0;
+        indexWave++;
+
+        emit WaveSetup(_waveType, _waveMaxTokens, _waveMaxTokensToBuy, _waveSingleTokenPrice);
+    }
+
+    function price(uint256 _count) public view virtual returns (uint256) {
+        return waveSingleTokenPrice * _count;
+    }
+
+    function chain() public view returns (uint256) {
+        return block.chainid;
+    }
+
+    function checkWaveNotComplete(uint256 _amount) internal view returns (bool) {
+        return _amount > 0 && waveTotalMinted + _amount <= waveMaxTokens;
+    }
+
+    function checkLimitNotReached(address _wallet, uint256 _amount) internal view returns (bool) {
+        return
+            waveOwnerToClaimedCounts[_wallet][indexWave - 1] + _amount <= waveMaxTokensToBuy &&
+            totalSupply() + _amount <= maxSupply;
+    }
+
+    function checkMintAllowed(address _wallet, uint256 _amount) public view returns (bool) {
+        return checkWaveNotComplete(_amount) && checkLimitNotReached(_wallet, _amount);
+    }
+
+    function mint(
+        address _wallet,
+        uint256 _amount,
+        uint256 _signatureId,
+        bytes memory _signature
+    ) external nonReentrant {
+        require(indexWave > 0, "Contract is not configured");
+        require(_msgSender() == allowedToExecuteMint, "Not allowed");
+        require(paused == 0, "Contract is paused");
+        require(_wallet != address(0x0), "Wallet is zero address");
+        require(_amount > 0, "Amount cannot be 0");
```

Comment:
_amount > 0 is also checked in the checkWaveNotCompleted at line 169 this could be removed